### ISSUE TITLE
C414: false positive

### ISSUE THREAD

**Issue by oliver-sanders** [adamchainz/flake8-comprehensions#661]
### Python Version

3.12.13

### flake8 Version

7.3.0

### Package Version

3.17.0

### Description

A false positive spotted with C414.

A sort wrapped within a sort is not necessarily redundant if the outer sort specifies a different sort key.

A multi-layer sort may be a valid approach for sorting by two fields, whether it's the nicest approach, however...

Example false positive

```python
x = sorted(sorted(y), key=len)  # C414 Unnecessary sorted call within sorted()
```

Demo:

```python
>>> with open('/usr/share/dict/words', 'r') as dictionary:  # linux example
...     words = dictionary.readlines()
... 
>>> slice = words[1000:1020]
>>> sorted(slice, key=len)
['Abp\n', 'abp\n', 'abow\n', 'abox\n', 'ABPC\n', 'aboves\n', 'abovesaid\n', 'aboveproof\n', 'above-said\n', 'above-found\n', 'above-given\n', 'aboveground\n', 'above-named\n', 'abovestairs\n', 'above-water\n', 'above-quoted\n', 'above-written\n', 'abovementioned\n', 'above-reported\n', 'above-mentioned\n']
>>> sorted(sorted(slice), key=len)
['Abp\n', 'abp\n', 'ABPC\n', 'abow\n', 'abox\n', 'aboves\n', 'abovesaid\n', 'above-said\n', 'aboveproof\n', 'above-found\n', 'above-given\n', 'above-named\n', 'above-water\n', 'aboveground\n', 'abovestairs\n', 'above-quoted\n', 'above-written\n', 'above-reported\n', 'abovementioned\n', 'above-mentioned\n']
>>> sorted(slice, key=len) == sorted(sorted(slice), key=len)
False
```
